tests/utils/fingerprints.py

```diff
@@ -34,9 +34,10 @@ def make_sparse_fp(
         Feature matrix.
     """
     vector_list = []
+    morgan_fp = rdkit_fp.GetMorganGenerator(radius=radius, fpSize=n_bits)
     for smiles in smiles_list:
         mol = Chem.MolFromSmiles(smiles)  # pylint: disable=no-member
-        vector = GetMorganFingerprintAsBitVect(mol, radius=radius, nBits=n_bits)
+        vector = morgan_fp.GetFingerprintAsNumPy(mol)
```

Comment:
The code is probably completely fine as is. Just a comment because maybe you had already experience with this: It could be worth doing an experiment to check if the explicit bit vector from `GetFingerprintAsNumpy` can be replaced by a call to `GetSparseFingerprint` which returns a `SparseBitVect` that is much smaller than the explicit version. In this way much less temporary memory would need to be allocated. However, I am not sure if the `csr_matrix` constructor supports the `SparseBitVect`. In addition, this more memory efficient way is probably slower than using numpy arrays.

Reply:
I have not done any experiments in this direction. I will transform this to a new issue so we keep it in mind.